docs/plugins/analysis-smartcn.asciidoc

```diff
@@ -17,7 +17,433 @@ include::install_remove.asciidoc[]
 [float]
 ==== `smartcn` tokenizer and token filter
 
-The plugin provides the `smartcn` analyzer and `smartcn_tokenizer` tokenizer,
-which are not configurable.
+The plugin provides the `smartcn` analyzer, `smartcn_tokenizer` tokenizer, and
+`smartcn_stop` token filter which are not configurable.
 
-NOTE: The `smartcn_word` token filter and `smartcn_sentence` have been deprecated.
+==== Reimplementing and extending the analyzers
+
+The `smartcn` analyzer could be reimplemented as a `custom` analyzer that can
+then be extended and configured as follows:
+
+[source,js]
+----------------------------------------------------
+PUT smartcn_example
+{
+  "settings": {
+    "analysis": {
+      "analyzer": {
+        "rebuilt_smartcn": {
+          "tokenizer":  "smartcn_tokenizer",
+          "filter": [
+            "porter_stem",
+            "smartcn_stop"
+          ]
+        }
+      }
+    }
+  }
+}
+----------------------------------------------------
+// CONSOLE
+// TEST[s/\n$/\nstartyaml\n  - compare_analyzers: {index: smartcn_example, first: smartcn, second: rebuilt_smartcn}\nendyaml\n/]
+
+[[analysis-smartcn_stop]]
+==== `smartcn_stop` token filter
+
+The `smartcn_stop` token filter filters out stopwords defined by `smartcn`
+analyzer (`_smartcn_`), and any other custom stopwords specified by the user.
+This filter only supports the predefined `_smartcn_` stopwords list.
+If you want to use a different predefined list, then use the
+{ref}/analysis-stop-tokenfilter.html[`stop` token filter] instead.
+
+[source,js]
+--------------------------------------------------
+PUT smartcn_example
+{
+  "settings": {
+    "index": {
+      "analysis": {
+        "analyzer": {
+          "smartcn_with_stop": {
+            "tokenizer": "smartcn_tokenizer",
+            "filter": [
+              "porter_stem",
+              "my_smartcn_stop"
+            ]
+          }
+        },
+        "filter": {
+          "my_smartcn_stop": {
+            "type": "smartcn_stop",
+            "stopwords": [
```

Comment:
since here are adding stopword filter example to the analyzer, maybe we can add a related stopword config related to the test text below to make it more reasonable, like add a stopword: `的` to this list. `的` is a Chinese modal particle, usually pointless.

Reply:
@medcl Thanks, I have addressed this in the second commit

Reply:
Thank you @mayya-sharipova